packages/data-broker/src/services/dhis/DhisService.js

```diff
@@ -226,27 +197,80 @@ export class DhisService extends Service {
     return buildAnalyticsFromEvents(translatedEvents, dataElements);
   };
 
-  pullAnalyticsFromEventsForApi = async (api, dataSources, options) => {
-    const { programCodes = [], period, startDate, endDate } = options;
+  translateDhisEventAnalytics = async (dhisAnalytics, dataSources) => {
     const dataElementCodes = dataSources.map(({ code }) => code);
-    const dhisElementCodes = dataSources.map(({ dataElementCode }) => dataElementCode);
 
-    const query = {
+    // Map to translation
+    const translatedEventAnalytics = await this.translator.translateInboundEventAnalytics(
+      dhisAnalytics,
+      dataSources,
+    );
+
+    // Map to analytics
+    const analytics = buildAnalyticsFromDhisEventAnalytics(
+      translatedEventAnalytics,
+      dataElementCodes,
+    );
+
+    return analytics;
+  };
+
+  getDataSourcesInProgram = async (dataSources, programCode) => {
+    const allowedDataSources = await this.models.dataSource.getDataElementsInGroup(programCode);
+    return dataSources.filter(({ id }) => allowedDataSources.includes(id));
+  };
+
+  mergeAnalytics = analytics => ({
+    results: analytics
+      .map(({ results }) => results)
+      .flat()
+      .sort(getSortByKey('period')),
+    metadata: {
+      dataElementCodeToName: analytics.reduce(
+        (dataElementCodeToName, { metadata }) => ({
+          ...dataElementCodeToName,
+          ...(metadata.dataElementCodeToName || {}),
+        }),
+        {},
+      ),
+    },
+  })
+
+  pullAnalyticsFromEventsForApi = async (api, dataSources, options) => {
+    const { programCodes = [], period, startDate, endDate, organisationUnitCodes } = options;
+
+    const baseQuery = {
       programCodes,
-      dataElementCodes: dhisElementCodes,
+      organisationUnitCodes,
```

Comment:
Will this have any unintended side affects?

Reply:
See [Nick's notes](https://github.com/beyondessential/tupaia/pull/1983#discussion_r575935785) on why it currently works and [Igor's addendum](https://github.com/beyondessential/tupaia/pull/2024#discussion_r576462039) on what the plan is, resulting in [this card](https://github.com/beyondessential/tupaia-backlog/issues/2251) to investigate further. Currently it looks like that card is a prerequisite for this one. Is that correct @IgorNadj?

Reply:
@IgorNadj  how does your investigation work affect the above?

Reply:
This issue has been resolved a while ago, see: https://github.com/beyondessential/tupaia-backlog/issues/2251